libopenage/pathfinding/integrator.cpp

```diff
@@ -36,24 +37,31 @@ std::shared_ptr<IntegrationField> Integrator::integrate(const std::shared_ptr<Co
                                                         sector_id_t other_sector_id,
                                                         const std::shared_ptr<Portal> &portal,
                                                         const coord::tile_delta &target,
-                                                        bool with_los) {
+                                                        bool with_los,
+                                                        bool evict_cache) {
 	auto cache_key = std::make_pair(portal->get_id(), other_sector_id);
-	auto cached = this->field_cache.find(cache_key);
-	if (cached != this->field_cache.end()) {
+	if (evict_cache) {
+		this->field_cache->evict(cache_key);
+	}
```

Comment:
Eviction should not be triggered by a parameter but by using a check of `cost_field->is_dirty()`